### PR TITLE
[Frontend] Add participant team approval feature by host

### PR DESCRIPTION
**Pull request by Suryansh5545**
This PR will allow the host to decide wheather he want to manually approve participant team and if yes then the host will be shown a Team Approved View as shown in the video below

https://github.com/Cloud-CV/EvalAI/assets/34577232/e2c35d02-ccad-48ef-b61f-bf307fd2f11d

### CODE REVIEW

**Review by gautamjajoo — CHANGES_REQUESTED** [review]
A couple of things:
1. The checkbox is placed weirdly. Can you add a similar margin on the card's left side and center the box?
2. a table scroll appears; attaching the screenshot as well.
3. The team created by the host should be automatically approved. @gchhablani, can you also share your opinion?
4. Can the tab `Team Approved` and the title `Team Approval` be the same? @gchhablani, what should be the name of the tab?
5. Can we approve the team in a similar way we verify submission on all submissions page?

File: apps/challenges/challenge_config_utils.py
Comment (gchhablani) [inline apps/challenges/challenge_config_utils.py:315]: Make this `manual_participant_approval` everywhere.

File: apps/challenges/migrations/0091_auto_manual_approval.py
Comment (gchhablani) [inline apps/challenges/migrations/0091_auto_manual_approval.py:10]: You will need to update the migration here. Delete this and generate migration again.

File: apps/challenges/views.py
Comment (gchhablani) [inline apps/challenges/views.py:591]: "Participant Team {name} was not approved"
"Participant Team {name} has existing submissions and cannot be unapproved"

^ Make these cases separate and use this text.

File: apps/challenges/views.py
Comment (Suryansh5545) [reply apps/challenges/views.py:400]: in getTeamName function in ChallengeCtrl. to verify if manual approval it on and if the team is approved or not. if not approved the submission will not be allowed to submitted

File: apps/challenges/views.py
Comment (Suryansh5545) [reply apps/challenges/views.py:591]: Done

File: frontend/src/js/controllers/challengeCtrl.js
Comment (gchhablani) [inline frontend/src/js/controllers/challengeCtrl.js:81]: The variable `approved_by_host` does not make sense. Not all challenges need approval from host. Make this variable `eligible_to_submit` or something.

File: frontend/src/js/controllers/challengeCtrl.js
Comment (gchhablani) [inline frontend/src/js/controllers/challengeCtrl.js:2923]: Remove this line.

File: frontend/src/views/web/challenge/edit-challenge/edit-challenge-approvalConfirm.html
Comment (gchhablani) [inline frontend/src/views/web/challenge/edit-challenge/edit-challenge-approvalConfirm.html:7]: <b>Note</b>: Participant teams with existing submissions cannot be disapproved

File: frontend/tests/controllers-test/challengeCtrl.test.js
Comment (gchhablani) [inline frontend/tests/controllers-test/challengeCtrl.test.js:636]: `eligible_to_submit`

File: frontend/src/views/web/challenge/submission.html
Comment (gchhablani) [inline frontend/src/views/web/challenge/submission.html:32]: `eligible_to_submit`

File: tests/unit/challenges/test_views.py
Comment (gchhablani) [inline tests/unit/challenges/test_views.py:236]: Excellent! Makes this a complete PR!

**Review by gchhablani — COMMENTED** [review]
This PR is ready to go!
Just added a few comments and questions.
Once those are resolved, we can merge and test it.

File: apps/challenges/views.py
Comment (Suryansh5545) [reply apps/challenges/views.py:348]: it is used by getTeamName in ChallengeCtl line 272 only

File: apps/challenges/views.py
Comment (Suryansh5545) [reply apps/challenges/views.py:399]: This endpoint is only used when challenge is manually approved, used to get the list of participant team and there approval status

File: apps/challenges/migrations/0093_auto_manual_approval.py
Comment (Suryansh5545) [reply apps/challenges/migrations/0093_auto_manual_approval.py]: Yes

**Pull request merged**
